### PR TITLE
feat: add hardhat config RPC methods

### PR DESCRIPTION
**Pull request by Wodann**
As part of this change, I made the `network_id` a `u64` as there was no need for it to support up to `U256`. It's represented as a string decimal in the JSON-RPC. The `U64` allows deserialization of these values too, making it a convenient helper.

### CODE REVIEW

File: crates/edr_rpc_hardhat/src/lib.rs [hunk 1]
```diff
@@ -118,7 +119,7 @@ pub enum Request {
         serialize_with = "single_to_sequence",
         deserialize_with = "sequence_to_single"
     )]
-    SetPrevRandao(ZeroXPrefixedBytes),
+    SetPrevRandao(B256),
```
Comment (agostbiro) [inline crates/edr_rpc_hardhat/src/lib.rs:122]: Just double checking: does `B256` serializes into `0x...` string?

File: crates/edr_provider/src/requests/hardhat/config.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    Ok(true)
+}
+
+pub fn handle_set_next_block_base_fee_per_gas_request(
+    data: &mut ProviderData,
+    base_fee_per_gas: U256,
+) -> Result<bool, ProviderError> {
+    let spec_id = data.spec_id();
+    if spec_id < SpecId::LONDON {
+        return Err(ProviderError::UnmetHardfork {
```
Comment (agostbiro) [inline crates/edr_provider/src/requests/hardhat/config.rs:32]: Wondering if this is going to match error returned by Hardhat. Is that a concern at all @alcuadrado @fvictorio ?

File: crates/edr_rpc_hardhat/src/metadata.rs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+/// Metadata about the forked network.
+#[derive(Clone, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ForkMetadata {
+    /// The chainId of the network that is being forked
+    pub chain_id: u64,
+    /// The number of the block that the network forked from.
+    pub fork_block_number: u64,
+    /// The hash of the block that the network forked from.
+    pub fork_block_hash: B256,
```
Comment (agostbiro) [inline crates/edr_rpc_hardhat/src/metadata.rs:36]: Noticed that the Hardhat type doesn't explicit say that this is a 0x-prefixed string. Are we sure it is?

```
    // The hash of the block that the network forked from.
    forkBlockHash: string;
```

**Review by agostbiro — APPROVED** [review]
Thanks LGTM overall. I left some questions to double check some things, approving assuming none of those need changes.

File: crates/edr_rpc_hardhat/src/lib.rs (hunk 1 above)
Comment (Wodann) [reply crates/edr_rpc_hardhat/src/lib.rs:122]: Yes, it does.

File: crates/edr_rpc_hardhat/src/metadata.rs (hunk 3 above)
Comment (Wodann) [reply crates/edr_rpc_hardhat/src/metadata.rs:36]:
```ts
    const latestBlockHashHex = latestBlock.header.hash().toString("hex");
    const latestBlockHash = `0x${latestBlockHashHex}`;
```
https://github.com/NomicFoundation/hardhat/blob/main/packages/hardhat-core/src/internal/hardhat-network/provider/node.ts#L2719

**Pull request merged**